frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -93,6 +94,7 @@ const TABLE_HEADER_HEIGHT = 45;
 function mapStateToProps(state) {
   return {
     metadata: getMetadata(state),
+    visualizationSettings: getVisualizationSettings(state),
```

Comment:
Rely on the computed viz settings. This handles missing columns and backfilling `table.columns` when needed.